api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -117,6 +118,32 @@ message ExtAuthz {
   // When this field is true, Envoy will include the peer X.509 certificate, if available, in the
   // :ref:`certificate<envoy_api_field_service.auth.v3.AttributeContext.Peer.certificate>`.
   bool include_peer_certificate = 10;
+
+  // Specifies when to enable the ext_authz filter.
+  // If specified, the ext_authz filter is enabled if the incoming request is matched. If not matched,
+  // the incoming request will be allowed as if the ext_authz filter doesn't exist (no authorization check request sent).
+  //
+  // For example, the following config enables ext_authz filter if the request has header "foo: bar"
+  // and the body includes the string "ext_authz".
+  //
+  // .. code-block:: yaml
+  //
+  //    match:
+  //      and_match:
+  //        rules:
+  //        - http_request_headers_match:
+  //            headers:
+  //            - name: "foo"
+  //              exact_match: "bar"
+  //        - http_request_generic_body_match:
+  //            patterns:
+  //            - string_match: "ext_authz"
+  //
+  // .. warning::
+  //
+  //   The `http_response_headers_match`, `http_response_trailers_match` and `http_response_generic_body_match`
+  //   rule is not supported.
+  config.common.matcher.v3.MatchPredicate match = 13;
```

Comment:
Sorry to put a wrench in this, and I should have seen this sooner, but I think this one-off is superseded by the proposal here from @snowp: https://github.com/envoyproxy/envoy/issues/12968. I would really rather not implement this when we are very close to having a better way doing this type of matching, as well as a generic way to do filter disabling.

How soon do you need this? I feel like we should try to intercept the @snowp proposal or have you help him with that proposal? WDYT? cc @htuch

Reply:
Thanks Matt. 

I just learned the proposal and commented in the design doc. I think the more general solution satisfies our main use case for ext_authz filter. We can discuss more tech details in the design doc.

The main concern is probably about the timeline. This PR is for the effort of better and first-class support of ext_authz filter planned in Istio 1.8 in November. We will need this PR early before Nov. because control plane changes are depending on this.

Reply:
OK. @snowp is out this week. Let's chat on Slack on Monday and try to find a path forward?

Reply:
yeah that sounds good to me, thanks.